src/datasources/cache/cache.first.data.source.ts

```diff
@@ -45,16 +48,36 @@ export class CacheFirstDataSource {
     const cached = await this.cacheService.get(cacheDir);
     if (cached != null) {
       winston.debug(`[Cache] Cache hit: [${cacheDir.key}, ${cacheDir.field}]`);
-      return JSON.parse(cached);
+      const data = JSON.parse(cached);
+      if (this.isNotFoundError(data)) {
+        throw new NetworkResponseError(404);
```

Comment:
Something to consider: if we cache 404 errors this way we just store the status code right? We loose any additional message that might come with the payload.

Reply:
I've changed it to store any additional data coming from the upstream service in [670d495](https://github.com/safe-global/safe-client-gateway-nest/pull/295/commits/670d49548487aa23e218fb2c3def0f8e70f0593a)